extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -209,20 +214,39 @@ export class AmpUserNotification extends AMP.BaseElement {
    * @return {!Promise}
    */
   postDismissEnpoint_() {
+    const enctype = this.element.getAttribute('enctype') ||
+      'application/json;charset=utf-8';
     return Services.xhrFor(this.win).fetchJson(
         dev().assertString(this.dismissHref_),
-        {
-          method: 'POST',
-          credentials: 'include',
-          requireAmpResponseSourceOrigin: false,
-          body: /** @type {!JsonObject} */({
-            'elementId': this.elementId_,
-            'ampUserId': this.ampUserId_,
-          }),
-          headers: {
-            'Content-Type': 'application/json;charset=utf-8',
-          },
-        });
+        this.buildPostDismissRequest_(enctype, this.elementId_,
+            this.ampUserId_));
+  }
+
+  /**
+   * Creates a Request to be used for postDismiss
+   * @private
+   * @param {!string} enctype
```

Comment:
nit: strings are already `!` so they don't need it. Only when nullable, then add `?`